projects/crud/src/lib/components/formset/formset.component.html

```diff
@@ -2,20 +2,22 @@
     <div class="row">
         <h4> {{ config.label }}</h4>
         <span class="spacer"></span>
-        <button mat-icon-button (click)="addForm()">
+        <button mat-icon-button (click)="addForm()" *ngIf="permissionsService.checkPermission(config?.permissions, permissionTypeEnum.update)">
```

Comment:
Can you convert the permissionsService to private in the container, and create a function in the component called checkPermissions that makes use of permissionsService ? it would be cleaner this way, and I was thinking maybe the permissions service in the future would return observables instead.